// angular2-toaster/src/toast.component.ngfactory.ts
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/render/api';
import * as import1 from '@angular/core/src/linker/view';
import * as import2 from '@angular/core/src/linker/element';
import * as import3 from './toast.component';
import * as import4 from '@angular/core/src/linker/view_utils';
import * as import5 from '@angular/core/src/di/injector';
import * as import6 from '@angular/core/src/linker/view_type';
import * as import7 from '@angular/core/src/change_detection/change_detection';
import * as import8 from '@angular/platform-browser/src/security/dom_sanitization_service';
import * as import9 from '@angular/core/src/linker/component_factory_resolver';
import * as import10 from '@angular/core/src/metadata/view';
import * as import11 from '@angular/core/src/linker/component_factory';
import * as import12 from '@angular/core/src/linker/query_list';
import * as import13 from '@angular/common/src/directives/ng_class';
import * as import14 from '@angular/common/src/directives/ng_switch';
import * as import15 from '@angular/common/src/directives/ng_if';
import * as import16 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import17 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import18 from '@angular/core/src/linker/element_ref';
import * as import19 from '@angular/core/src/linker/template_ref';
import * as import20 from '@angular/core/src/security';
var renderType_ToastComponent_Host:import0.RenderComponentType = (null as any);
class _View_ToastComponent_Host0 extends import1.AppView<any> {
  _el_0:any;
  /*private*/ _appEl_0:import2.AppElement;
  _ToastComponent_0_4:import3.ToastComponent;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import2.AppElement) {
    super(_View_ToastComponent_Host0,renderType_ToastComponent_Host,import6.ViewType.HOST,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import2.AppElement {
    this._el_0 = this.selectOrCreateHostElement('div',rootSelector,(null as any));
    this.renderer.setElementAttribute(this._el_0,'toastComp','');
    this._appEl_0 = new import2.AppElement(0,(null as any),this,this._el_0);
    var compView_0:any = viewFactory_ToastComponent0(this.viewUtils,this.injector(0),this._appEl_0);
    this._ToastComponent_0_4 = new import3.ToastComponent(this.parentInjector.get(import8.DomSanitizer),this.parentInjector.get(import9.ComponentFactoryResolver),compView_0.ref);
    this._appEl_0.initComponent(this._ToastComponent_0_4,[],compView_0);
    compView_0.create(this._ToastComponent_0_4,this.projectableNodes,(null as any));
    this.init([].concat([this._el_0]),[this._el_0],[],[]);
    return this._appEl_0;
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import3.ToastComponent) && (0 === requestNodeIndex))) { return this._ToastComponent_0_4; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    if (((this.numberOfChecks === 0) && !throwOnChange)) { this._ToastComponent_0_4.ngOnInit(); }
    this.detectContentChildrenChanges(throwOnChange);
    this.detectViewChildrenChanges(throwOnChange);
    if (!throwOnChange) { if ((this.numberOfChecks === 0)) { this._ToastComponent_0_4.ngAfterViewInit(); } }
  }
}
function viewFactory_ToastComponent_Host0(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import2.AppElement):import1.AppView<any> {
  if ((renderType_ToastComponent_Host === (null as any))) { (renderType_ToastComponent_Host = viewUtils.createRenderComponentType('',0,import10.ViewEncapsulation.None,[],{})); }
  return new _View_ToastComponent_Host0(viewUtils,parentInjector,declarationEl);
}
export const ToastComponentNgFactory:import11.ComponentFactory<import3.ToastComponent> = new import11.ComponentFactory<import3.ToastComponent>('[toastComp]',viewFactory_ToastComponent_Host0,import3.ToastComponent);
const styles_ToastComponent:any[] = [];
var renderType_ToastComponent:import0.RenderComponentType = (null as any);
class _View_ToastComponent0 extends import1.AppView<import3.ToastComponent> {
  _viewQuery_componentBody_0:import12.QueryList<any>;
  _text_0:any;
  _el_1:any;
  _NgClass_1_3:import13.NgClass;
  _text_2:any;
  _el_3:any;
  _text_4:any;
  _el_5:any;
  _NgClass_5_3:import13.NgClass;
  _text_6:any;
  _text_7:any;
  _el_8:any;
  _NgClass_8_3:import13.NgClass;
  _NgSwitch_8_4:import14.NgSwitch;
  _text_9:any;
  _anchor_10:any;
  /*private*/ _appEl_10:import2.AppElement;
  _TemplateRef_10_5:any;
  _NgSwitchCase_10_6:import14.NgSwitchCase;
  _text_11:any;
  _anchor_12:any;
  /*private*/ _appEl_12:import2.AppElement;
  _TemplateRef_12_5:any;
  _NgSwitchCase_12_6:import14.NgSwitchCase;
  _text_13:any;
  _anchor_14:any;
  /*private*/ _appEl_14:import2.AppElement;
  _TemplateRef_14_5:any;
  _NgSwitchCase_14_6:import14.NgSwitchCase;
  _text_15:any;
  _text_16:any;
  _text_17:any;
  _anchor_18:any;
  /*private*/ _appEl_18:import2.AppElement;
  _TemplateRef_18_5:any;
  _NgIf_18_6:import15.NgIf;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  /*private*/ _expr_5:any;
  /*private*/ _expr_6:any;
  /*private*/ _expr_7:any;
  /*private*/ _expr_8:any;
  /*private*/ _expr_9:any;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import2.AppElement) {
    super(_View_ToastComponent0,renderType_ToastComponent,import6.ViewType.COMPONENT,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import2.AppElement {
    const parentRenderNode:any = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
    this._viewQuery_componentBody_0 = new import12.QueryList<any>();
    this._text_0 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._el_1 = this.renderer.createElement(parentRenderNode,'i',(null as any));
    this.renderer.setElementAttribute(this._el_1,'class','toaster-icon');
    this._NgClass_1_3 = new import13.NgClass(this.parentInjector.get(import16.IterableDiffers),this.parentInjector.get(import17.KeyValueDiffers),new import18.ElementRef(this._el_1),this.renderer);
    this._text_2 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._el_3 = this.renderer.createElement(parentRenderNode,'div',(null as any));
    this.renderer.setElementAttribute(this._el_3,'class','toast-content');
    this._text_4 = this.renderer.createText(this._el_3,'\n            ',(null as any));
    this._el_5 = this.renderer.createElement(this._el_3,'div',(null as any));
    this._NgClass_5_3 = new import13.NgClass(this.parentInjector.get(import16.IterableDiffers),this.parentInjector.get(import17.KeyValueDiffers),new import18.ElementRef(this._el_5),this.renderer);
    this._text_6 = this.renderer.createText(this._el_5,'',(null as any));
    this._text_7 = this.renderer.createText(this._el_3,'\n            ',(null as any));
    this._el_8 = this.renderer.createElement(this._el_3,'div',(null as any));
    this._NgClass_8_3 = new import13.NgClass(this.parentInjector.get(import16.IterableDiffers),this.parentInjector.get(import17.KeyValueDiffers),new import18.ElementRef(this._el_8),this.renderer);
    this._NgSwitch_8_4 = new import14.NgSwitch();
    this._text_9 = this.renderer.createText(this._el_8,'\n                ',(null as any));
    this._anchor_10 = this.renderer.createTemplateAnchor(this._el_8,(null as any));
    this._appEl_10 = new import2.AppElement(10,8,this,this._anchor_10);
    this._TemplateRef_10_5 = new import19.TemplateRef_(this._appEl_10,viewFactory_ToastComponent1);
    this._NgSwitchCase_10_6 = new import14.NgSwitchCase(this._appEl_10.vcRef,this._TemplateRef_10_5,this._NgSwitch_8_4);
    this._text_11 = this.renderer.createText(this._el_8,'\n                ',(null as any));
    this._anchor_12 = this.renderer.createTemplateAnchor(this._el_8,(null as any));
    this._appEl_12 = new import2.AppElement(12,8,this,this._anchor_12);
    this._TemplateRef_12_5 = new import19.TemplateRef_(this._appEl_12,viewFactory_ToastComponent2);
    this._NgSwitchCase_12_6 = new import14.NgSwitchCase(this._appEl_12.vcRef,this._TemplateRef_12_5,this._NgSwitch_8_4);
    this._text_13 = this.renderer.createText(this._el_8,'\n                ',(null as any));
    this._anchor_14 = this.renderer.createTemplateAnchor(this._el_8,(null as any));
    this._appEl_14 = new import2.AppElement(14,8,this,this._anchor_14);
    this._TemplateRef_14_5 = new import19.TemplateRef_(this._appEl_14,viewFactory_ToastComponent3);
    this._NgSwitchCase_14_6 = new import14.NgSwitchCase(this._appEl_14.vcRef,this._TemplateRef_14_5,this._NgSwitch_8_4);
    this._text_15 = this.renderer.createText(this._el_8,'\n            ',(null as any));
    this._text_16 = this.renderer.createText(this._el_3,'\n        ',(null as any));
    this._text_17 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._anchor_18 = this.renderer.createTemplateAnchor(parentRenderNode,(null as any));
    this._appEl_18 = new import2.AppElement(18,(null as any),this,this._anchor_18);
    this._TemplateRef_18_5 = new import19.TemplateRef_(this._appEl_18,viewFactory_ToastComponent4);
    this._NgIf_18_6 = new import15.NgIf(this._appEl_18.vcRef,this._TemplateRef_18_5);
    this._expr_0 = import7.UNINITIALIZED;
    this._expr_1 = import7.UNINITIALIZED;
    this._expr_2 = import7.UNINITIALIZED;
    this._expr_3 = import7.UNINITIALIZED;
    this._expr_4 = import7.UNINITIALIZED;
    this._expr_5 = import7.UNINITIALIZED;
    this._expr_6 = import7.UNINITIALIZED;
    this._expr_7 = import7.UNINITIALIZED;
    this._expr_8 = import7.UNINITIALIZED;
    this._expr_9 = import7.UNINITIALIZED;
    this.init([],[
      this._text_0,
      this._el_1,
      this._text_2,
      this._el_3,
      this._text_4,
      this._el_5,
      this._text_6,
      this._text_7,
      this._el_8,
      this._text_9,
      this._anchor_10,
      this._text_11,
      this._anchor_12,
      this._text_13,
      this._anchor_14,
      this._text_15,
      this._text_16,
      this._text_17,
      this._anchor_18
    ]
    ,[],[]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import13.NgClass) && (1 === requestNodeIndex))) { return this._NgClass_1_3; }
    if (((token === import13.NgClass) && ((5 <= requestNodeIndex) && (requestNodeIndex <= 6)))) { return this._NgClass_5_3; }
    if (((token === import19.TemplateRef) && (10 === requestNodeIndex))) { return this._TemplateRef_10_5; }
    if (((token === import14.NgSwitchCase) && (10 === requestNodeIndex))) { return this._NgSwitchCase_10_6; }
    if (((token === import19.TemplateRef) && (12 === requestNodeIndex))) { return this._TemplateRef_12_5; }
    if (((token === import14.NgSwitchCase) && (12 === requestNodeIndex))) { return this._NgSwitchCase_12_6; }
    if (((token === import19.TemplateRef) && (14 === requestNodeIndex))) { return this._TemplateRef_14_5; }
    if (((token === import14.NgSwitchCase) && (14 === requestNodeIndex))) { return this._NgSwitchCase_14_6; }
    if (((token === import13.NgClass) && ((8 <= requestNodeIndex) && (requestNodeIndex <= 15)))) { return this._NgClass_8_3; }
    if (((token === import14.NgSwitch) && ((8 <= requestNodeIndex) && (requestNodeIndex <= 15)))) { return this._NgSwitch_8_4; }
    if (((token === import19.TemplateRef) && (18 === requestNodeIndex))) { return this._TemplateRef_18_5; }
    if (((token === import15.NgIf) && (18 === requestNodeIndex))) { return this._NgIf_18_6; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0:any = 'toaster-icon';
    if (import4.checkBinding(throwOnChange,this._expr_0,currVal_0)) {
      this._NgClass_1_3.klass = currVal_0;
      this._expr_0 = currVal_0;
    }
    const currVal_1:any = this.context.iconClass;
    if (import4.checkBinding(throwOnChange,this._expr_1,currVal_1)) {
      this._NgClass_1_3.ngClass = currVal_1;
      this._expr_1 = currVal_1;
    }
    if (!throwOnChange) { this._NgClass_1_3.ngDoCheck(); }
    const currVal_2:any = this.context.toast.toasterConfig.titleClass;
    if (import4.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this._NgClass_5_3.ngClass = currVal_2;
      this._expr_2 = currVal_2;
    }
    if (!throwOnChange) { this._NgClass_5_3.ngDoCheck(); }
    const currVal_4:any = this.context.toast.toasterConfig.messageClass;
    if (import4.checkBinding(throwOnChange,this._expr_4,currVal_4)) {
      this._NgClass_8_3.ngClass = currVal_4;
      this._expr_4 = currVal_4;
    }
    if (!throwOnChange) { this._NgClass_8_3.ngDoCheck(); }
    const currVal_5:any = this.context.toast.bodyOutputType;
    if (import4.checkBinding(throwOnChange,this._expr_5,currVal_5)) {
      this._NgSwitch_8_4.ngSwitch = currVal_5;
      this._expr_5 = currVal_5;
    }
    const currVal_6:any = this.context.bodyOutputType.Component;
    if (import4.checkBinding(throwOnChange,this._expr_6,currVal_6)) {
      this._NgSwitchCase_10_6.ngSwitchCase = currVal_6;
      this._expr_6 = currVal_6;
    }
    const currVal_7:any = this.context.bodyOutputType.TrustedHtml;
    if (import4.checkBinding(throwOnChange,this._expr_7,currVal_7)) {
      this._NgSwitchCase_12_6.ngSwitchCase = currVal_7;
      this._expr_7 = currVal_7;
    }
    const currVal_8:any = this.context.bodyOutputType.Default;
    if (import4.checkBinding(throwOnChange,this._expr_8,currVal_8)) {
      this._NgSwitchCase_14_6.ngSwitchCase = currVal_8;
      this._expr_8 = currVal_8;
    }
    const currVal_9:any = this.context.toast.showCloseButton;
    if (import4.checkBinding(throwOnChange,this._expr_9,currVal_9)) {
      this._NgIf_18_6.ngIf = currVal_9;
      this._expr_9 = currVal_9;
    }
    this.detectContentChildrenChanges(throwOnChange);
    const currVal_3:any = import4.interpolate(1,'',this.context.toast.title,'');
    if (import4.checkBinding(throwOnChange,this._expr_3,currVal_3)) {
      this.renderer.setText(this._text_6,currVal_3);
      this._expr_3 = currVal_3;
    }
    this.detectViewChildrenChanges(throwOnChange);
    if (!throwOnChange) { if (this._viewQuery_componentBody_0.dirty) {
        this._viewQuery_componentBody_0.reset([this._appEl_10.mapNestedViews(_View_ToastComponent1,(nestedView:_View_ToastComponent1):any => {
          return [nestedView._appEl_0.vcRef];
      })]);
      this.context.componentBody = this._viewQuery_componentBody_0.first;
    } }
  }
}
export function viewFactory_ToastComponent0(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import2.AppElement):import1.AppView<import3.ToastComponent> {
  if ((renderType_ToastComponent === (null as any))) { (renderType_ToastComponent = viewUtils.createRenderComponentType('',0,import10.ViewEncapsulation.None,styles_ToastComponent,{})); }
  return new _View_ToastComponent0(viewUtils,parentInjector,declarationEl);
}
class _View_ToastComponent1 extends import1.AppView<any> {
  _el_0:any;
  /*private*/ _appEl_0:import2.AppElement;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import2.AppElement) {
    super(_View_ToastComponent1,renderType_ToastComponent,import6.ViewType.EMBEDDED,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import2.AppElement {
    this._el_0 = this.renderer.createElement((null as any),'div',(null as any));
    this._appEl_0 = new import2.AppElement(0,(null as any),this,this._el_0);
    this.init([].concat([this._appEl_0]),[this._el_0],[],[]);
    return (null as any);
  }
  dirtyParentQueriesInternal():void {
    (<_View_ToastComponent0>this.parent)._viewQuery_componentBody_0.setDirty();
  }
}
function viewFactory_ToastComponent1(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import2.AppElement):import1.AppView<any> {
  return new _View_ToastComponent1(viewUtils,parentInjector,declarationEl);
}
class _View_ToastComponent2 extends import1.AppView<any> {
  _el_0:any;
  /*private*/ _expr_0:any;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import2.AppElement) {
    super(_View_ToastComponent2,renderType_ToastComponent,import6.ViewType.EMBEDDED,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import2.AppElement {
    this._el_0 = this.renderer.createElement((null as any),'div',(null as any));
    this._expr_0 = import7.UNINITIALIZED;
    this.init([].concat([this._el_0]),[this._el_0],[],[]);
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this.detectContentChildrenChanges(throwOnChange);
    const currVal_0:any = this.parent.context.toast.body;
    if (import4.checkBinding(throwOnChange,this._expr_0,currVal_0)) {
      this.renderer.setElementProperty(this._el_0,'innerHTML',this.viewUtils.sanitizer.sanitize(import20.SecurityContext.HTML,currVal_0));
      this._expr_0 = currVal_0;
    }
    this.detectViewChildrenChanges(throwOnChange);
  }
}
function viewFactory_ToastComponent2(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import2.AppElement):import1.AppView<any> {
  return new _View_ToastComponent2(viewUtils,parentInjector,declarationEl);
}
class _View_ToastComponent3 extends import1.AppView<any> {
  _el_0:any;
  _text_1:any;
  /*private*/ _expr_0:any;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import2.AppElement) {
    super(_View_ToastComponent3,renderType_ToastComponent,import6.ViewType.EMBEDDED,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import2.AppElement {
    this._el_0 = this.renderer.createElement((null as any),'div',(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'',(null as any));
    this._expr_0 = import7.UNINITIALIZED;
    this.init([].concat([this._el_0]),[
      this._el_0,
      this._text_1
    ]
    ,[],[]);
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this.detectContentChildrenChanges(throwOnChange);
    const currVal_0:any = import4.interpolate(1,'',this.parent.context.toast.body,'');
    if (import4.checkBinding(throwOnChange,this._expr_0,currVal_0)) {
      this.renderer.setText(this._text_1,currVal_0);
      this._expr_0 = currVal_0;
    }
    this.detectViewChildrenChanges(throwOnChange);
  }
}
function viewFactory_ToastComponent3(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import2.AppElement):import1.AppView<any> {
  return new _View_ToastComponent3(viewUtils,parentInjector,declarationEl);
}
class _View_ToastComponent4 extends import1.AppView<any> {
  _el_0:any;
  _text_1:any;
  /*private*/ _expr_1:any;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import2.AppElement) {
    super(_View_ToastComponent4,renderType_ToastComponent,import6.ViewType.EMBEDDED,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import2.AppElement {
    this._el_0 = this.renderer.createElement((null as any),'div',(null as any));
    this.renderer.setElementAttribute(this._el_0,'class','toast-close-button');
    this._text_1 = this.renderer.createText(this._el_0,'\n        ',(null as any));
    this._expr_1 = import7.UNINITIALIZED;
    var disposable_0:Function = this.renderer.listen(this._el_0,'click',this.eventHandler(this._handle_click_0_0.bind(this)));
    this.init([].concat([this._el_0]),[
      this._el_0,
      this._text_1
    ]
    ,[disposable_0],[]);
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this.detectContentChildrenChanges(throwOnChange);
    const currVal_1:any = this.parent.context.safeCloseHtml;
    if (import4.checkBinding(throwOnChange,this._expr_1,currVal_1)) {
      this.renderer.setElementProperty(this._el_0,'innerHTML',this.viewUtils.sanitizer.sanitize(import20.SecurityContext.HTML,currVal_1));
      this._expr_1 = currVal_1;
    }
    this.detectViewChildrenChanges(throwOnChange);
  }
  private _handle_click_0_0($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_0:any = ((<any>this.parent.context.click($event,this.parent.context.toast)) !== false);
    return (true && pd_0);
  }
}
function viewFactory_ToastComponent4(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import2.AppElement):import1.AppView<any> {
  return new _View_ToastComponent4(viewUtils,parentInjector,declarationEl);
}
