// angular2-toaster/src/toaster-container.component.spec.ngfactory.ts
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from './toaster-container.component.spec';
import * as import2 from '@angular/common/src/common_module';
import * as import3 from './toaster.module';
import * as import4 from '@angular/common/src/localization';
import * as import5 from './toaster.service';
import * as import6 from '@angular/core/src/di/injector';
import * as import7 from '@angular/core/src/i18n/tokens';
import * as import8 from '@angular/core/src/render/api';
import * as import9 from '@angular/core/src/linker/view';
import * as import10 from '@angular/core/src/linker/element';
import * as import11 from '@angular/core/src/linker/view_utils';
import * as import12 from '@angular/core/src/linker/view_type';
import * as import13 from '@angular/core/src/change_detection/change_detection';
import * as import14 from '@angular/core/src/metadata/view';
import * as import15 from '@angular/core/src/linker/component_factory';
import * as import16 from './toaster-container.component';
import * as import17 from './toaster-container.component.ngfactory';
class TestComponentModuleInjector extends import0.NgModuleInjector<import1.TestComponentModule> {
  _CommonModule_0:import2.CommonModule;
  _ToasterModule_1:import3.ToasterModule;
  _TestComponentModule_2:import1.TestComponentModule;
  __NgLocalization_3:import4.NgLocaleLocalization;
  __ToasterService_4:import5.ToasterService;
  constructor(parent:import6.Injector) {
    super(parent,[],[]);
  }
  get _NgLocalization_3():import4.NgLocaleLocalization {
    if ((this.__NgLocalization_3 == (null as any))) { (this.__NgLocalization_3 = new import4.NgLocaleLocalization(this.parent.get(import7.LOCALE_ID))); }
    return this.__NgLocalization_3;
  }
  get _ToasterService_4():import5.ToasterService {
    if ((this.__ToasterService_4 == (null as any))) { (this.__ToasterService_4 = new import5.ToasterService()); }
    return this.__ToasterService_4;
  }
  createInternal():import1.TestComponentModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._ToasterModule_1 = new import3.ToasterModule();
    this._TestComponentModule_2 = new import1.TestComponentModule();
    return this._TestComponentModule_2;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.ToasterModule)) { return this._ToasterModule_1; }
    if ((token === import1.TestComponentModule)) { return this._TestComponentModule_2; }
    if ((token === import4.NgLocalization)) { return this._NgLocalization_3; }
    if ((token === import5.ToasterService)) { return this._ToasterService_4; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const TestComponentModuleNgFactory:import0.NgModuleFactory<import1.TestComponentModule> = new import0.NgModuleFactory(TestComponentModuleInjector,import1.TestComponentModule);
class TestDynamicComponentModuleInjector extends import0.NgModuleInjector<import1.TestDynamicComponentModule> {
  _TestDynamicComponentModule_0:import1.TestDynamicComponentModule;
  constructor(parent:import6.Injector) {
    super(parent,[TestDynamicComponentNgFactory],[TestDynamicComponentNgFactory]);
  }
  createInternal():import1.TestDynamicComponentModule {
    this._TestDynamicComponentModule_0 = new import1.TestDynamicComponentModule();
    return this._TestDynamicComponentModule_0;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import1.TestDynamicComponentModule)) { return this._TestDynamicComponentModule_0; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const TestDynamicComponentModuleNgFactory:import0.NgModuleFactory<import1.TestDynamicComponentModule> = new import0.NgModuleFactory(TestDynamicComponentModuleInjector,import1.TestDynamicComponentModule);
class TestBoundDynamicComponentModuleInjector extends import0.NgModuleInjector<import1.TestBoundDynamicComponentModule> {
  _TestBoundDynamicComponentModule_0:import1.TestBoundDynamicComponentModule;
  constructor(parent:import6.Injector) {
    super(parent,[TestBoundDynamicComponentNgFactory],[TestBoundDynamicComponentNgFactory]);
  }
  createInternal():import1.TestBoundDynamicComponentModule {
    this._TestBoundDynamicComponentModule_0 = new import1.TestBoundDynamicComponentModule();
    return this._TestBoundDynamicComponentModule_0;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import1.TestBoundDynamicComponentModule)) { return this._TestBoundDynamicComponentModule_0; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const TestBoundDynamicComponentModuleNgFactory:import0.NgModuleFactory<import1.TestBoundDynamicComponentModule> = new import0.NgModuleFactory(TestBoundDynamicComponentModuleInjector,import1.TestBoundDynamicComponentModule);
var renderType_TestComponent_Host:import8.RenderComponentType = (null as any);
class _View_TestComponent_Host0 extends import9.AppView<any> {
  _el_0:any;
  /*private*/ _appEl_0:import10.AppElement;
  _TestComponent_0_4:import1.TestComponent;
  constructor(viewUtils:import11.ViewUtils,parentInjector:import6.Injector,declarationEl:import10.AppElement) {
    super(_View_TestComponent_Host0,renderType_TestComponent_Host,import12.ViewType.HOST,viewUtils,parentInjector,declarationEl,import13.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import10.AppElement {
    this._el_0 = this.selectOrCreateHostElement('test-component',rootSelector,(null as any));
    this._appEl_0 = new import10.AppElement(0,(null as any),this,this._el_0);
    var compView_0:any = viewFactory_TestComponent0(this.viewUtils,this.injector(0),this._appEl_0);
    this._TestComponent_0_4 = new import1.TestComponent(this.parentInjector.get(import5.ToasterService));
    this._appEl_0.initComponent(this._TestComponent_0_4,[],compView_0);
    compView_0.create(this._TestComponent_0_4,this.projectableNodes,(null as any));
    this.init([].concat([this._el_0]),[this._el_0],[],[]);
    return this._appEl_0;
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import1.TestComponent) && (0 === requestNodeIndex))) { return this._TestComponent_0_4; }
    return notFoundResult;
  }
}
function viewFactory_TestComponent_Host0(viewUtils:import11.ViewUtils,parentInjector:import6.Injector,declarationEl:import10.AppElement):import9.AppView<any> {
  if ((renderType_TestComponent_Host === (null as any))) { (renderType_TestComponent_Host = viewUtils.createRenderComponentType('',0,import14.ViewEncapsulation.None,[],{})); }
  return new _View_TestComponent_Host0(viewUtils,parentInjector,declarationEl);
}
export const TestComponentNgFactory:import15.ComponentFactory<import1.TestComponent> = new import15.ComponentFactory<import1.TestComponent>('test-component',viewFactory_TestComponent_Host0,import1.TestComponent);
const styles_TestComponent:any[] = [];
var renderType_TestComponent:import8.RenderComponentType = (null as any);
class _View_TestComponent0 extends import9.AppView<import1.TestComponent> {
  _el_0:any;
  /*private*/ _appEl_0:import10.AppElement;
  _ToasterContainerComponent_0_4:import16.ToasterContainerComponent;
  /*private*/ _expr_0:any;
  constructor(viewUtils:import11.ViewUtils,parentInjector:import6.Injector,declarationEl:import10.AppElement) {
    super(_View_TestComponent0,renderType_TestComponent,import12.ViewType.COMPONENT,viewUtils,parentInjector,declarationEl,import13.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import10.AppElement {
    const parentRenderNode:any = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
    this._el_0 = this.renderer.createElement(parentRenderNode,'toaster-container',(null as any));
    this._appEl_0 = new import10.AppElement(0,(null as any),this,this._el_0);
    var compView_0:any = import17.viewFactory_ToasterContainerComponent0(this.viewUtils,this.injector(0),this._appEl_0);
    this._ToasterContainerComponent_0_4 = new import16.ToasterContainerComponent(this.parentInjector.get(import5.ToasterService),compView_0.ref);
    this._appEl_0.initComponent(this._ToasterContainerComponent_0_4,[],compView_0);
    compView_0.create(this._ToasterContainerComponent_0_4,[],(null as any));
    this._expr_0 = import13.UNINITIALIZED;
    this.init([],[this._el_0],[],[]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import16.ToasterContainerComponent) && (0 === requestNodeIndex))) { return this._ToasterContainerComponent_0_4; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0:any = this.context.toasterconfig;
    if (import11.checkBinding(throwOnChange,this._expr_0,currVal_0)) {
      this._ToasterContainerComponent_0_4.toasterconfig = currVal_0;
      this._expr_0 = currVal_0;
    }
    if (((this.numberOfChecks === 0) && !throwOnChange)) { this._ToasterContainerComponent_0_4.ngOnInit(); }
    this.detectContentChildrenChanges(throwOnChange);
    this.detectViewChildrenChanges(throwOnChange);
  }
  destroyInternal():void {
    this._ToasterContainerComponent_0_4.ngOnDestroy();
  }
}
export function viewFactory_TestComponent0(viewUtils:import11.ViewUtils,parentInjector:import6.Injector,declarationEl:import10.AppElement):import9.AppView<import1.TestComponent> {
  if ((renderType_TestComponent === (null as any))) { (renderType_TestComponent = viewUtils.createRenderComponentType('',0,import14.ViewEncapsulation.None,styles_TestComponent,{})); }
  return new _View_TestComponent0(viewUtils,parentInjector,declarationEl);
}
var renderType_TestDynamicComponent_Host:import8.RenderComponentType = (null as any);
class _View_TestDynamicComponent_Host0 extends import9.AppView<any> {
  _el_0:any;
  /*private*/ _appEl_0:import10.AppElement;
  _TestDynamicComponent_0_4:import1.TestDynamicComponent;
  constructor(viewUtils:import11.ViewUtils,parentInjector:import6.Injector,declarationEl:import10.AppElement) {
    super(_View_TestDynamicComponent_Host0,renderType_TestDynamicComponent_Host,import12.ViewType.HOST,viewUtils,parentInjector,declarationEl,import13.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import10.AppElement {
    this._el_0 = this.selectOrCreateHostElement('test-dynamic-component',rootSelector,(null as any));
    this._appEl_0 = new import10.AppElement(0,(null as any),this,this._el_0);
    var compView_0:any = viewFactory_TestDynamicComponent0(this.viewUtils,this.injector(0),this._appEl_0);
    this._TestDynamicComponent_0_4 = new import1.TestDynamicComponent();
    this._appEl_0.initComponent(this._TestDynamicComponent_0_4,[],compView_0);
    compView_0.create(this._TestDynamicComponent_0_4,this.projectableNodes,(null as any));
    this.init([].concat([this._el_0]),[this._el_0],[],[]);
    return this._appEl_0;
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import1.TestDynamicComponent) && (0 === requestNodeIndex))) { return this._TestDynamicComponent_0_4; }
    return notFoundResult;
  }
}
function viewFactory_TestDynamicComponent_Host0(viewUtils:import11.ViewUtils,parentInjector:import6.Injector,declarationEl:import10.AppElement):import9.AppView<any> {
  if ((renderType_TestDynamicComponent_Host === (null as any))) { (renderType_TestDynamicComponent_Host = viewUtils.createRenderComponentType('',0,import14.ViewEncapsulation.None,[],{})); }
  return new _View_TestDynamicComponent_Host0(viewUtils,parentInjector,declarationEl);
}
export const TestDynamicComponentNgFactory:import15.ComponentFactory<import1.TestDynamicComponent> = new import15.ComponentFactory<import1.TestDynamicComponent>('test-dynamic-component',viewFactory_TestDynamicComponent_Host0,import1.TestDynamicComponent);
const styles_TestDynamicComponent:any[] = [];
var renderType_TestDynamicComponent:import8.RenderComponentType = (null as any);
class _View_TestDynamicComponent0 extends import9.AppView<import1.TestDynamicComponent> {
  _el_0:any;
  _text_1:any;
  constructor(viewUtils:import11.ViewUtils,parentInjector:import6.Injector,declarationEl:import10.AppElement) {
    super(_View_TestDynamicComponent0,renderType_TestDynamicComponent,import12.ViewType.COMPONENT,viewUtils,parentInjector,declarationEl,import13.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import10.AppElement {
    const parentRenderNode:any = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
    this._el_0 = this.renderer.createElement(parentRenderNode,'div',(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'loaded via component',(null as any));
    this.init([],[
      this._el_0,
      this._text_1
    ]
    ,[],[]);
    return (null as any);
  }
}
export function viewFactory_TestDynamicComponent0(viewUtils:import11.ViewUtils,parentInjector:import6.Injector,declarationEl:import10.AppElement):import9.AppView<import1.TestDynamicComponent> {
  if ((renderType_TestDynamicComponent === (null as any))) { (renderType_TestDynamicComponent = viewUtils.createRenderComponentType('',0,import14.ViewEncapsulation.None,styles_TestDynamicComponent,{})); }
  return new _View_TestDynamicComponent0(viewUtils,parentInjector,declarationEl);
}
var renderType_TestBoundDynamicComponent_Host:import8.RenderComponentType = (null as any);
class _View_TestBoundDynamicComponent_Host0 extends import9.AppView<any> {
  _el_0:any;
  /*private*/ _appEl_0:import10.AppElement;
  _TestBoundDynamicComponent_0_4:import1.TestBoundDynamicComponent;
  constructor(viewUtils:import11.ViewUtils,parentInjector:import6.Injector,declarationEl:import10.AppElement) {
    super(_View_TestBoundDynamicComponent_Host0,renderType_TestBoundDynamicComponent_Host,import12.ViewType.HOST,viewUtils,parentInjector,declarationEl,import13.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import10.AppElement {
    this._el_0 = this.selectOrCreateHostElement('bound-dynamic-component',rootSelector,(null as any));
    this._appEl_0 = new import10.AppElement(0,(null as any),this,this._el_0);
    var compView_0:any = viewFactory_TestBoundDynamicComponent0(this.viewUtils,this.injector(0),this._appEl_0);
    this._TestBoundDynamicComponent_0_4 = new import1.TestBoundDynamicComponent();
    this._appEl_0.initComponent(this._TestBoundDynamicComponent_0_4,[],compView_0);
    compView_0.create(this._TestBoundDynamicComponent_0_4,this.projectableNodes,(null as any));
    this.init([].concat([this._el_0]),[this._el_0],[],[]);
    return this._appEl_0;
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import1.TestBoundDynamicComponent) && (0 === requestNodeIndex))) { return this._TestBoundDynamicComponent_0_4; }
    return notFoundResult;
  }
}
function viewFactory_TestBoundDynamicComponent_Host0(viewUtils:import11.ViewUtils,parentInjector:import6.Injector,declarationEl:import10.AppElement):import9.AppView<any> {
  if ((renderType_TestBoundDynamicComponent_Host === (null as any))) { (renderType_TestBoundDynamicComponent_Host = viewUtils.createRenderComponentType('',0,import14.ViewEncapsulation.None,[],{})); }
  return new _View_TestBoundDynamicComponent_Host0(viewUtils,parentInjector,declarationEl);
}
export const TestBoundDynamicComponentNgFactory:import15.ComponentFactory<import1.TestBoundDynamicComponent> = new import15.ComponentFactory<import1.TestBoundDynamicComponent>('bound-dynamic-component',viewFactory_TestBoundDynamicComponent_Host0,import1.TestBoundDynamicComponent);
const styles_TestBoundDynamicComponent:any[] = [];
var renderType_TestBoundDynamicComponent:import8.RenderComponentType = (null as any);
class _View_TestBoundDynamicComponent0 extends import9.AppView<import1.TestBoundDynamicComponent> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  /*private*/ _expr_0:any;
  constructor(viewUtils:import11.ViewUtils,parentInjector:import6.Injector,declarationEl:import10.AppElement) {
    super(_View_TestBoundDynamicComponent0,renderType_TestBoundDynamicComponent,import12.ViewType.COMPONENT,viewUtils,parentInjector,declarationEl,import13.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import10.AppElement {
    const parentRenderNode:any = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
    this._el_0 = this.renderer.createElement(parentRenderNode,'div',(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'',(null as any));
    this._el_2 = this.renderer.createElement(this._el_0,'button',(null as any));
    this.renderer.setElementAttribute(this._el_2,'id','click');
    this._expr_0 = import13.UNINITIALIZED;
    var disposable_0:Function = this.renderer.listen(this._el_2,'click',this.eventHandler(this._handle_click_2_0.bind(this)));
    this.init([],[
      this._el_0,
      this._text_1,
      this._el_2
    ]
    ,[disposable_0],[]);
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this.detectContentChildrenChanges(throwOnChange);
    const currVal_0:any = import11.interpolate(1,'',this.context.someValue,' loaded via component');
    if (import11.checkBinding(throwOnChange,this._expr_0,currVal_0)) {
      this.renderer.setText(this._text_1,currVal_0);
      this._expr_0 = currVal_0;
    }
    this.detectViewChildrenChanges(throwOnChange);
  }
  private _handle_click_2_0($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_0:any = ((<any>this.context.clickHandler()) !== false);
    return (true && pd_0);
  }
}
export function viewFactory_TestBoundDynamicComponent0(viewUtils:import11.ViewUtils,parentInjector:import6.Injector,declarationEl:import10.AppElement):import9.AppView<import1.TestBoundDynamicComponent> {
  if ((renderType_TestBoundDynamicComponent === (null as any))) { (renderType_TestBoundDynamicComponent = viewUtils.createRenderComponentType('',0,import14.ViewEncapsulation.None,styles_TestBoundDynamicComponent,{})); }
  return new _View_TestBoundDynamicComponent0(viewUtils,parentInjector,declarationEl);
}
